src/mainwindow.cpp

```diff
@@ -69,93 +101,111 @@ void MainWindow::handleReloadPorts() {
     loadPortsAndSet();
 }
 
-inline bool MainWindow::tryOpen() {
-    if (!m_monitor->m_serialPort.isOpen()) {
-        m_monitor->openingFlag = true;
-        emit open();
-        while (m_monitor->openingFlag);
-        if (!m_monitor->m_serialPort.isOpen()) return false;
-    }
-    return true;
-}
-
-void MainWindow::handleMonitorToggle(bool checked) {
+void MainWindow::handleMonitorToggled(bool checked) {
     if (checked) {
-        if (tryOpen()) {
-            emit startMonitor();
-        }
+        startMonitor();
     } else {
         stopMonitor();
     }
 }
 
-void MainWindow::handlePlotter(bool checked) {
+void MainWindow::handleReadyRead() {
+    // the first pass generally has corrupted data, so clear the serial port's buffer
+    if (m_readFirstPass) {
+        m_readFirstPass = false;
+        m_serialPort.clear();
+        return;
+    }
+    QByteArray buf = m_serialPort.readAll();
+    if (buf.length() > 0) {
+        emit sendToWorker(buf);
+    }
+}
+
+void MainWindow::handlePlotterToggled(bool checked) {
     if (checked) {
         m_plotterView = new PlotterView;
         m_plotterView->move(x() + 10 + width(), y());
         m_plotterView->show();
+        worker->plotEnabled = true;
+        connect(worker, &Worker::plotPoint, m_plotterView, &PlotterView::plotPoint);
```

Comment:
Do you need to create a new `PlotterView` (and destroy it) every time the plotter is toggled? 

If possible, might be better to do something like `m_plotterView.toggle()` so that less Qt overhead is incurred.

Reply:
Does Qt actually let you do that with dialog boxes? I doubt

Reply:
`toggle` would be something you implement